### ISSUE TITLE
Failed building with "Could not find module Text.Regex.TDFA"

### ISSUE THREAD

**Issue by hai-nguyen-van** [koalaman/shellcheck#2280]
I cannot build shellcheck either from `master` or `v0.7.1`. After a fresh install of `cabal-install` on my Ubuntu 20.04, I have cloned the `shellcheck` repo adn then executed these commands in the repo. I get the following errors:
```bash
$ cabal update
Warning: The update command is a part of the legacy v1 style of cabal usage.

Please switch to using either the new project style and the new-update command
or the legacy v1-update alias as new-style projects will become the default in
the next version of cabal-install. Please file a bug if you cannot replicate a
working v1- use case with the new-style commands.

For more information, see: https://wiki.haskell.org/Cabal/NewBuild

Config file path source is default config file.
Config file /home/nguyen-van/.cabal/config not found.
Writing default configuration to /home/nguyen-van/.cabal/config
Downloading the latest package list from hackage.haskell.org
```

```bash
$ cabal install
Warning: The install command is a part of the legacy v1 style of cabal usage.

Please switch to using either the new project style and the new-install
command or the legacy v1-install alias as new-style projects will become the
default in the next version of cabal-install. Please file a bug if you cannot
replicate a working v1- use case with the new-style commands.

For more information, see: https://wiki.haskell.org/Cabal/NewBuild

Resolving dependencies...
Starting     ShellCheck-0.7.1
Building     ShellCheck-0.7.1
Failed to install ShellCheck-0.7.1
Build log ( /home/nguyen-van/.cabal/logs/ghc-8.6.5/ShellCheck-0.7.1-9DYhuIxkVJi7Lpwjgsx6lb.log ):
cabal: Entering directory '.'
Configuring ShellCheck-0.7.1...
Preprocessing library for ShellCheck-0.7.1..
Building library for ShellCheck-0.7.1..
[ 1 of 23] Compiling ShellCheck.Regex ( src/ShellCheck/Regex.hs, dist/build/ShellCheck/Regex.o )

src/ShellCheck/Regex.hs:28:1: error:
    Could not find module ‘Text.Regex.TDFA’
    There are files missing in the ‘regex-tdfa-1.3.1.1’ package,
    try running 'ghc-pkg check'.
    Use -v to see a list of the files searched for.
   |
28 | import Text.Regex.TDFA
   | ^^^^^^^^^^^^^^^^^^^^^^
cabal: Leaving directory '.'
cabal: Error: some packages failed to install:
ShellCheck-0.7.1-9DYhuIxkVJi7Lpwjgsx6lb failed during the building phase. The
exception was:
ExitFailure 1

⇀ ⊣ nguyen-van@hai-thinkpad-x1-carbon: ~/Desktop/shellcheck ((HEAD detached at v0.7.1))
$ cabal install ShellCheck-0.7.1
Warning: The install command is a part of the legacy v1 style of cabal usage.

Please switch to using either the new project style and the new-install
command or the legacy v1-install alias as new-style projects will become the
default in the next version of cabal-install. Please file a bug if you cannot
replicate a working v1- use case with the new-style commands.

For more information, see: https://wiki.haskell.org/Cabal/NewBuild

Resolving dependencies...
Downloading  ShellCheck-0.7.1
Downloaded   ShellCheck-0.7.1
Starting     ShellCheck-0.7.1
Building     ShellCheck-0.7.1
Failed to install ShellCheck-0.7.1
Build log ( /home/nguyen-van/.cabal/logs/ghc-8.6.5/ShellCheck-0.7.1-9DYhuIxkVJi7Lpwjgsx6lb.log ):
cabal: Entering directory '/tmp/cabal-tmp-54947/ShellCheck-0.7.1'
Configuring ShellCheck-0.7.1...
Preprocessing library for ShellCheck-0.7.1..
Building library for ShellCheck-0.7.1..
[ 1 of 23] Compiling Paths_ShellCheck ( dist/build/autogen/Paths_ShellCheck.hs, dist/build/Paths_ShellCheck.o )
[ 2 of 23] Compiling ShellCheck.Regex ( src/ShellCheck/Regex.hs, dist/build/ShellCheck/Regex.o )

src/ShellCheck/Regex.hs:28:1: error:
    Could not find module ‘Text.Regex.TDFA’
    There are files missing in the ‘regex-tdfa-1.3.1.1’ package,
    try running 'ghc-pkg check'.
    Use -v to see a list of the files searched for.
   |
28 | import Text.Regex.TDFA
   | ^^^^^^^^^^^^^^^^^^^^^^
cabal: Leaving directory '/tmp/cabal-tmp-54947/ShellCheck-0.7.1'
cabal: Error: some packages failed to install:
ShellCheck-0.7.1-9DYhuIxkVJi7Lpwjgsx6lb failed during the building phase. The
exception was:
ExitFailure 1

⇀ ⊣ nguyen-van@hai-thinkpad-x1-carbon: ~/Desktop/shellcheck ((HEAD detached at v0.7.1))
$ git checkout master
Previous HEAD position was f7547c9 Stable version v0.7.1
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
→ ⊣ nguyen-van@hai-thinkpad-x1-carbon: ~/Desktop/shellcheck (master)
$ cabal install 
Warning: The install command is a part of the legacy v1 style of cabal usage.

Please switch to using either the new project style and the new-install
command or the legacy v1-install alias as new-style projects will become the
default in the next version of cabal-install. Please file a bug if you cannot
replicate a working v1- use case with the new-style commands.

For more information, see: https://wiki.haskell.org/Cabal/NewBuild

Resolving dependencies...
In order, the following will be installed:
ShellCheck-0.7.2 (reinstall)
Warning: Note that reinstalls are always dangerous. Continuing anyway...
Starting     ShellCheck-0.7.2
Building     ShellCheck-0.7.2
Failed to install ShellCheck-0.7.2
Build log ( /home/nguyen-van/.cabal/logs/ghc-8.6.5/ShellCheck-0.7.2-K1PXGYHexmf4YjGV4axfYE.log ):
cabal: Entering directory '.'
Configuring ShellCheck-0.7.2...
Preprocessing library for ShellCheck-0.7.2..
Building library for ShellCheck-0.7.2..
[ 1 of 23] Compiling ShellCheck.Regex ( src/ShellCheck/Regex.hs, dist/build/ShellCheck/Regex.o ) [Text.Regex.TDFA changed]

src/ShellCheck/Regex.hs:28:1: error:
    Could not find module ‘Text.Regex.TDFA’
    There are files missing in the ‘regex-tdfa-1.3.1.1’ package,
    try running 'ghc-pkg check'.
    Use -v to see a list of the files searched for.
   |
28 | import Text.Regex.TDFA
   | ^^^^^^^^^^^^^^^^^^^^^^
cabal: Leaving directory '.'
cabal: Error: some packages failed to install:
ShellCheck-0.7.2-K1PXGYHexmf4YjGV4axfYE failed during the building phase. The
exception was:
ExitFailure 1
```

**Comment by koalaman** [comment]
I'm able to successfully build with them on Ubuntu 20.04, here tested via Docker:

```
FROM ubuntu:20.04
RUN apt-get update && apt-get install -y cabal-install git
RUN git clone --branch v0.7.1 https://github.com/koalaman/shellcheck
RUN cabal update
RUN cd shellcheck && cabal install
```

* Your output says there are missing files and suggests running `ghc-pkg check`. Did you try that? 
* You can clear out your per-user Haskell libs and executables with `rm -rf ~/.cabal ~/.ghc` and try again 
* If still no luck, try updating `ghc` and `cabal-install`. The latest versions on Ubuntu 20.04 appear to be 8.8.1 and 2.4.0.0 respectively, while you're on ghc-8.6.5. Cabal especially changes quite rapidly.

Please reopen if this does not help!

**Comment by hai-nguyen-van** [comment]
Thanks. I only deleted `.cabal` and forgot `.ghc`. This solved the problem.
